app/routes/postsRouter.ts

```diff
@@ -1,76 +1,69 @@
-            post.topic = req.body.topic;
-            post.post = req.body.post;
-            post.datetime = new Date(Date.now());
-            post.categories = req.body.categoryId;
-            post.pinned = req.body.pinned;
-            
-            let validateErrors: ValidationError[] = await validate(post);
-            
-            if(validateErrors.length > 0) {
-                errors = ehandler.handleValidationErrors(validateErrors);
-                throw 400;
-            }
-            let postRepository = getRepository(Post);
-            await postRepository.save(post);
-            
-            res.status(200).send(post);
-        } catch(error) {
-            if (error === 400) {
-                res.status(400).send(errors)
-            } else {
-                res.status(400).send({message: "Couldn't save the data", "Error": error.message});
-            }
-        }
-        
-        
-            
-        
-    }
-    //Modify a post in database
-    public update(req: Request, res: Response, next: NextFunction) {
-        res.send("Nothing");
+  //get all posts from database
+  public getAll = async(req: Request, res: Response, next: NextFunction) => {
+    try {
+      let results = await this.postRepo.find();
+      res.status(200).send(results);
+    } catch (error) {
+        console.log(error);
+      res.status(400).send({ message: "Couldn't get the data", Error: error });
     }
-    //Delete a post from database
-    public delete(req: Request, res: Response, next: NextFunction) {
-        res.send("Nothing");
+  }
+  //get specific post from database
+  public getOne(req: Request, res: Response, next: NextFunction) {
+    res.send('Nothing');
+  }
+  //Add new post to database
+  public async create(req: Request, res: Response, next: NextFunction) {
+    let ehandler = new ErrorHandler();
+    let errors;
+    try {
+      let post = new Post();
+      post.topic = req.body.topic;
+      post.post = req.body.post;
+      post.datetime = new Date(Date.now());
+      post.categories = req.body.categoryId;
+      post.pinned = req.body.pinned;
+
+      let validateErrors: ValidationError[] = await validate(post);
+
+      if (validateErrors.length > 0) {
+        errors = ehandler.handleValidationErrors(validateErrors);
+        throw 400;
+      }
+      let postRepository = this.postRepo;
+      await postRepository.save(post);
+
+      res.status(200).send(post);
+    } catch (error) {
+      if (error === 400) {
+        res.status(400).send(errors);
+      } else {
+        res.status(400).send({ message: "Couldn't save the data", Error: error.message });
+      }
     }
-    
-    
+  }
+  //Modify a post in database
+  public update(req: Request, res: Response, next: NextFunction) {
+    res.send('Nothing');
+  }
+  //Delete a post from database
+  public delete(req: Request, res: Response, next: NextFunction) {
+    res.send('Nothing');
+  }
 }
 
-
 export default PostsRouter;
```

Comment:
Unify the exports of classes. Because in some file we are using `export default` and some times `export`. Any of them is good. Just pick one ;)